### PR TITLE
fix(search): fix on expanded attributes filtering #238

### LINKED ISSUE #238: Get Temporal evolution : only temporal attributes of the entity must be returned by default

### PR DESCRIPTION
**Pull request by philippestoltz**

### CODE REVIEW

File: search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityAttributeService.kt [hunk 1]
```diff
@@ -145,11 +144,7 @@ class TemporalEntityAttributeService(
             WHERE temporal_entity_attribute.entity_id = :entity_id
             """.trimIndent()
 
-        val expandedAttrsList =
-            attrs.map {
-                expandJsonLdKey(it, contextLink)!!
```
Comment (bobeal) [inline search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityAttributeService.kt:150]: the `contextLink` param of this method should no longer be used.

File: search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt [hunk 2]
```diff
@@ -184,8 +190,16 @@ internal fun buildTemporalQuery(params: MultiValueMap<String, String>): Temporal
...
+        ?.split(",")
+        .orEmpty()
+        .map {
+            JsonLdUtils.expandJsonLdKey(it, contextLink)!!
+        }
+        ?.toSet()
+
     return TemporalQuery(
-        attrs = attrsParam?.split(",")?.toSet().orEmpty(),
+        attrs = expandedAttrs,
```
Comment (bobeal) [inline search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt:202]: for clarity, you could rename this attribute to `expandedAttrs`

File: search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt [hunk 3]
```diff
@@ -113,11 +113,17 @@ class TemporalEntityHandler(
         else
             attributeAndResultsMap.keys.map { it.attributeName }.toSet()
 
-        val compactedJsonLdEntity = JsonLdUtils.filterCompactedEntityOnAttributes(
-            jsonLdEntityWithTemporalValues.compact(),
-            attributesToFilter
+        jsonLdEntityWithTemporalValues.properties.toMutableMap()
```
Comment (bobeal) [inline search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt:116]: unused?

File: search-service/src/test/kotlin/com/egm/stellio/search/web/TemporalEntityHandlerTests.kt [hunk 4]
```diff
@@ -46,6 +46,8 @@ class TemporalEntityHandlerTests {
     @Value("\${application.jsonld.apic_context}")
     val apicContext: String? = null
 
+    val context = "https://raw.githubusercontent.com/easy-global-market/ngsild-api-data-models/" +
```
Comment (bobeal) [inline search-service/src/test/kotlin/com/egm/stellio/search/web/TemporalEntityHandlerTests.kt:49]: isn't it the same as the above `apicContext`?

File: shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt [hunk 5]
```diff
@@ -1,5 +1,7 @@
 package com.egm.stellio.shared.util
 
+import arrow.core.extensions.list.monoidal.identity
+import arrow.core.identity
```
Comment (bobeal) [inline shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt:4]: those 2 imports are not used

File: shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt [hunk 6]
```diff
@@ -256,12 +258,29 @@ object JsonLdUtils {
...
+    ): Map<String, Any> {
+        val identity: (CompactedJsonLdEntity) -> CompactedJsonLdEntity = { it }
+        return filterEntityOnAttributes(input, identity, includedAttributes)
+    }
+
+    fun filterJsonLdEntityOnAttributes(
+        input: JsonLdEntity,
+        includedAttributes: Set<String>
+    ): Map<String, Any> {
+        val inputToMap = { i: JsonLdEntity -> i.properties.toMap() }
```
Comment (bobeal) [inline shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt:270]: is not `i.properties` already a map?

File: search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityAttributeService.kt (hunk 1 above)
Comment (philippestoltz) [reply search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityAttributeService.kt:150]: of course, strange that my IDE does not notify that

File: search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt (hunk 2 above)
Comment (philippestoltz) [reply search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt:202]: ok

File: search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt (hunk 3 above)
Comment (philippestoltz) [reply search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt:116]: yep, sorry

File: search-service/src/test/kotlin/com/egm/stellio/search/web/TemporalEntityHandlerTests.kt (hunk 4 above)
Comment (philippestoltz) [reply search-service/src/test/kotlin/com/egm/stellio/search/web/TemporalEntityHandlerTests.kt:49]: yes

File: shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt (hunk 5 above)
Comment (philippestoltz) [reply shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt:4]: yes, weird that detekt did not detect it

File: shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt (hunk 6 above)
Comment (philippestoltz) [reply shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt:270]: of course

File: search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt [hunk 7]
```diff
@@ -184,8 +187,16 @@ internal fun buildTemporalQuery(params: MultiValueMap<String, String>): Temporal
         if (it >= 1) it else null
     }
 
+    val expandedAttrs = attrsParam
+        ?.split(",")
+        .orEmpty()
+        .map {
+            JsonLdUtils.expandJsonLdKey(it, contextLink)!!
+        }
+        ?.toSet()
```
Comment (HoucemKacem) [inline search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt:196]: I think it's unecessary safe call.

**Review by HoucemKacem — APPROVED** [review]

File: search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt (hunk 7 above)
Comment (philippestoltz) [reply search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt:196]: you're right

**Review by bobeal — APPROVED** [review]

**Pull request merged**